### ISSUE TITLE
FindClose receive a Handle type different than the returned by FindFirstFile

### ISSUE THREAD

**Issue by bsonnino** [microsoft/CsWin32#240]
When you start a file enumeration with FindFirstFile. it returns a handle of type FindCloseSafeHandle, but this type is not compatible with FindClose, which receives a handle of type HANDLE. So this code:
```C#
var handle = PInvoke.FindFirstFile("*.*", out var findData);
bool result;
do
{
    result = PInvoke.FindNextFile(handle, out findData);
} while (result);
PInvoke.FindClose(handle);
```
Doesn't compile. An alternative to it would be:
``` C#
using var handle = PInvoke.FindFirstFile("*.*", out var findData);
bool result;
do
{
    result = PInvoke.FindNextFile(handle, out findData);
} while (result);
```
But this is not as explicit as using FindClose

**Comment by jnm2** [comment]
`using` is the gold standard for this. You want the handle to be freed if an exception is thrown before the end of the loop.

**Comment by bsonnino** [comment]
Yes, I know that, but it's weird to have a FindCloseSafeHandle that cannot be used in FindClose, no. Besides that, I would use something like this, as it's more explicit:
```C#
var handle = PInvoke.FindFirstFile("*.*", out var findData);
try
{
  bool result;
  do
  {
      result = PInvoke.FindNextFile(handle, out findData);
  } while (result);
}
finally
{
  PInvoke.FindClose(handle);
}
```

**Comment by jnm2** [comment]
I think it would be weirder to not follow the longstanding .NET convention for scoping a resource with `using` or by explicitly calling `.Dispose()`. I've thought it was strange that handle-closing methods were generated in `PInvoke` in the first place. Maybe @AArnott can clue us in.

**Comment by AArnott** [comment]
There appears to be a bug in the metadata in this area, but perhaps not the one you thought of.

In the metadata, `FindFirstFile` returns `FindFileHandle` (a typedef struct) while `FindClose` takes `HANDLE`. **`FindClose` should take the same handle type that is returned by `FindFirstFile`.**

With CsWin32, that means that the `extern` method overload for `FindFirstFile` will match `FindClose` and you can write the code you want if you're calling the `extern` method overload. But if you are calling the method overload that returns a `SafeHandle`, you are expected to dispose of the result *rather* than call `FindClose` explicitly.

I'm moving the bug to the metadata repo to track fixing the `FindClose` parameter type.
